### PR TITLE
Makes artguns unswappable

### PR DESCRIPTION
**Pull request by ThatFiveGuys**
## About the PR 
Title

## Why's this needed? 
Artcells in artguns are balanced for the most part, however in eguns or sec weaponry they behave as basically rampage weaponry, thanks to the insane self recharge of artcells. This pr makes artguns unable to have there cell swapped.

## Changelog

```changelog
(u)Ikea
(*)Artgun cells are now unswappable.
```

### CODE REVIEW

File: code/obj/artifacts/artifact_items/energy_gun.dm [hunk 1]
```diff
@@ -8,6 +8,7 @@
 	is_syndicate = 1
 	mat_changename = 0
 	mat_changedesc = 0
+	var/can_swap_cell = 0
```
Comment (Zonespace27) [inline code/obj/artifacts/artifact_items/energy_gun.dm:11]:
```suggestion
	can_swap_cell = 0
```

**Comment by frawhst** [comment]
Note that this doesn't stop you getting the cells, as melting or otherwise destroying the gun also drops the cell.

File: code/obj/artifacts/artifact_items/energy_gun.dm (hunk 1 above)
Comment (ThatFiveGuys) [reply code/obj/artifacts/artifact_items/energy_gun.dm:11]: Oh yeah turns out doing a lazy copy and paste and not double checking is a bad.

**Comment by ThatFiveGuys** [comment]
> Note that this doesn't stop you getting the cells, as melting or otherwise destroying the gun also drops the cell.

Explosions dont work, acid does.

**Comment by ThatFiveGuys** [comment]
> Note that this doesn't stop you getting the cells, as melting or otherwise destroying the gun also drops the cell.

Fixed, thanks for alerting me of this.

**Comment by Zonespace27** [comment]
Maybe slightly out of scope, but changing the file to use True/False instead of 1/0 would be nice QOL since you're working on it

**Comment by zjdtmkhzt** [comment]
I'd be a little sad to see artifact cells practically gone by being merged with the guns, but it's not entirely untrue that they can be very rampage-enabling.

If this goes through, we should maybe consider raising the floor on them to 100 charge instead of 50, otherwise you could have guns with cells that do not allow even firing a single shot, which would be kind of a bummer if you can't swap out the cell.

An alternative nerf could be maybe giving the cells a similar treatment to guns, where prolonged usage gives the cell increasing amounts of faults, which would affect the user. I think that could be a fun way to make abusing alien cells more risky, while still allowing people to make that decision.

(Also, I second what zonespace said, please use TRUE/FALSE instead of numbers.)

**Comment by ThatFiveGuys** [comment]
Raising floor is far from a bad idea, also true/false would be good however id have to change every single gun if i wanted to properly swap it over. Ill look into lava moon and vardulak though.

**Comment by unfunnyperson** [comment]
> An alternative nerf could be maybe giving the cells a similar treatment to guns, where prolonged usage gives the cell increasing amounts of faults, which would affect the user. I think that could be a fun way to make abusing alien cells more risky, while still allowing people to make that decision.
I like this a lot more than just removing the ability to get artcells from guns.

**Comment by ThatFiveGuys** [comment]
> > An alternative nerf could be maybe giving the cells a similar treatment to guns, where prolonged usage gives the cell increasing amounts of faults, which would affect the user. I think that could be a fun way to make abusing alien cells more risky, while still allowing people to make that decision.
> 
> I like this a lot more than just removing the ability to get artcells from guns.

Wouldnt really be able to do it in a way thatd sufficently nerf it I fell. The upside of a weapon with practically infinite charge is a lot, and like, I cant think of sufficent downsides of faults thatd make the faults not worth using the artcell.

**Comment by zjdtmkhzt** [comment]
The faults are already coded, you don't need to do anything for that except make it call the proc that rolls a chance to add an artifact fault to the cell. You can check out the various faults in faults.dm

As for the TRUE/FALSE thing, you do not need to change all guns. Writing TRUE/FALSE is basically the same as writing 1/0 in this case, the compiler just replaces it. It is for readability of the code.

**Comment by ThatFiveGuys** [comment]
Decided to go with faults after some fault, closing this for now.

**Comment by zjdtmkhzt** [comment]
Didn't need to close this, could have just changed the code here too, since there is history here in the comments about the decision making, and it's not much code change. But either works!

**Pull request closed without merging**